src/Simplex/Messaging/Agent.hs

```diff
@@ -1979,6 +2102,28 @@ processSMPTransmission c@AgentClient {smpClients, subQ} (tSess@(_, srv, _), v, s
           | internalPrevMsgHash /= receivedPrevMsgHash = MsgError MsgBadHash
           | otherwise = MsgError MsgDuplicate -- this case is not possible
 
+withSwitchedRQ :: AgentMonad m => RcvQueue -> RcvSwitchStatus -> m a -> m a
```

Comment:
This is bad decomposition. It does two things - checks the condition, and then calls actions. What was needed is a function internalErr, and two lines in all places where it is called. And I am not even sure it should be INTERNAL error

Reply:
what was the point of passing action here and not just calling it outside?